src/foam/dao/index/AltIndex.java

```diff
@@ -55,22 +57,31 @@ public Object remove(Object state, FObject value) {
   public Object removeAll() {
     Object[] s = toObjectArray(null);
 
-    for ( int i = 0 ; i < delegates_.size() ; i++ )
+    for (int i = 0; i < delegates_.size(); i++)
       s[i] = delegates_.get(i).removeAll();
 
     return s;
   }
 
+  public FObject find(Object state, Object key) {
+    FindPlan plan = this.planFind(state, key);
+    Object[] s = (Object[]) state;
+    if (s == null || s.length == 0) {
+      return null;
+    }
+    return plan.find(s[0], key);
+  }
+
   public FindPlan planFind(Object state, Object key) {
     Object[] s = toObjectArray(state);
     Plan bestPlan = NoPlan.instance();
 
-    for ( int i = 0 ; i < delegates_.size() ; i++ ) {
+    for (int i = 0; i < delegates_.size(); i++) {
```

Comment:
Here and on other lines you've broken the FOAM spacing guideline:

https://github.com/foam-framework/foam2/blob/master/doc/guides/StyleGuide.md